### PR TITLE
Do not set empty environment variables

### PR DESCRIPTION
**Pull request by zc-devs**
Related #1911, #1913.

While #1913 fixes manual trigger and numbers/booleans for automatic triggers, it pollutes Pod's environment with empty strings:
```
        - name: CI_STEP_NAME
        - name: CI_WORKFLOW_NUMBER
          value: '1'
        - name: CI_COMMIT_TARGET_BRANCH
        - name: CI_COMMIT_REFSPEC
        - name: CI_STEP_NUMBER
          value: '0'
        - name: CI_COMMIT_PULL_REQUEST
```

### CODE REVIEW

**Comment by zc-devs** [comment]
Before #1913, default values (empty string, zero, false) are filtered. It caused confusing, when I run pipeline manually. So, I removed it entirely. Now, there are a lot of empty string vars.
I think filtering zeros and falses is a mistake, because it is valid value. If `nothing`/`not set` needs to be expressed, null pointer should be used in the model.
However, it is a little bit different for strings, IMO. What's the meaning of empty string? The same as null, if we are talking about the model. Therefore this PR.

This PR should not break manual trigger, because manual envs is a _different_ map, and this PR filters only empty strings in transition from _model_ to envs.
Besides, looking at #1913 tests, there's no test with empty value. They are filtered in UI, as I remember correctly. So, the same logic, as in this PR.

File: docs/docs/20-usage/50-environment.md [hunk 1]
```diff
@@ -68,11 +68,11 @@ This is the reference list of all environment variables available to your pipeli
...
 | `CI_COMMIT_BRANCH`               | commit branch (equals target branch for pull requests)                                                             | `main`                                                                                     |
-| `CI_COMMIT_SOURCE_BRANCH`        | commit source branch (empty if event is not `pull_request` or `pull_request_closed`)                               | `issue-branch`                                                                             |
-| `CI_COMMIT_TARGET_BRANCH`        | commit target branch (empty if event is not `pull_request` or `pull_request_closed`)                               | `main`                                                                                     |
+| `CI_COMMIT_SOURCE_BRANCH`        | commit source branch (sets only for `pull_request` and `pull_request_closed` events)                               | `issue-branch`                                                                             |
```
Comment (pat-s) [inline docs/docs/20-usage/50-environment.md:71]:
```suggestion
| `CI_COMMIT_SOURCE_BRANCH`        | commit source branch (set only for `pull_request` and `pull_request_closed` events)                               | `issue-branch`                                                                             |
```

File: docs/docs/20-usage/50-environment.md [hunk 2]
```diff
@@ -68,11 +68,11 @@ This is the reference list of all environment variables available to your pipeli
...
+| `CI_COMMIT_TARGET_BRANCH`        | commit target branch (sets only for `pull_request` and `pull_request_closed` events)                               | `main`                                                                                     |
```
Comment (pat-s) [inline docs/docs/20-usage/50-environment.md:72]:
```suggestion
| `CI_COMMIT_TARGET_BRANCH`        | commit target branch (set only for `pull_request` and `pull_request_closed` events)                               | `main`                                                                                     |
```

File: docs/docs/20-usage/50-environment.md [hunk 3]
```diff
@@ -68,11 +68,11 @@ This is the reference list of all environment variables available to your pipeli
...
 | `CI_COMMIT_TAG`                  | commit tag name (empty if event is not `tag`)                                                                      | `v1.10.3`                                                                                  |
-| `CI_COMMIT_PULL_REQUEST`         | commit pull request number (empty if event is not `pull_request` or `pull_request_closed`)                         | `1`                                                                                        |
-| `CI_COMMIT_PULL_REQUEST_LABELS`  | labels assigned to pull request (empty if event is not `pull_request` or `pull_request_closed`)                    | `server`                                                                                   |
+| `CI_COMMIT_PULL_REQUEST`         | commit pull request number (sets only for `pull_request` and `pull_request_closed` events)                         | `1`                                                                                        |
```
Comment (pat-s) [inline docs/docs/20-usage/50-environment.md:74]:
```suggestion
| `CI_COMMIT_PULL_REQUEST`         | commit pull request number (set only for `pull_request` and `pull_request_closed` events)                         | `1`                                                                                        |
```

File: docs/docs/20-usage/50-environment.md [hunk 4]
```diff
@@ -68,11 +68,11 @@ This is the reference list of all environment variables available to your pipeli
...
+| `CI_COMMIT_PULL_REQUEST_LABELS`  | labels assigned to pull request (sets only for `pull_request` and `pull_request_closed` events)                    | `server`                                                                                   |
```
Comment (pat-s) [inline docs/docs/20-usage/50-environment.md:75]:
```suggestion
| `CI_COMMIT_PULL_REQUEST_LABELS`  | labels assigned to pull request (set only for `pull_request` and `pull_request_closed` events)                    | `server`                                                                                   |
```

File: docs/docs/20-usage/50-environment.md [hunk 5]
```diff
@@ -101,8 +101,8 @@ This is the reference list of all environment variables available to your pipeli
...
 | `CI_PREV_COMMIT_BRANCH`          | previous commit branch                                                                                             | `main`                                                                                     |
-| `CI_PREV_COMMIT_SOURCE_BRANCH`   | previous commit source branch                                                                                      | `issue-branch`                                                                             |
-| `CI_PREV_COMMIT_TARGET_BRANCH`   | previous commit target branch                                                                                      | `main`                                                                                     |
+| `CI_PREV_COMMIT_SOURCE_BRANCH`   | previous commit source branch (sets only for `pull_request` and `pull_request_closed` events)                      | `issue-branch`                                                                             |
```
Comment (pat-s) [inline docs/docs/20-usage/50-environment.md:104]:
```suggestion
| `CI_PREV_COMMIT_SOURCE_BRANCH`   | previous commit source branch (set only for `pull_request` and `pull_request_closed` events)                      | `issue-branch`                                                                             |
```

File: docs/docs/20-usage/50-environment.md [hunk 6]
```diff
@@ -101,8 +101,8 @@ This is the reference list of all environment variables available to your pipeli
...
+| `CI_PREV_COMMIT_TARGET_BRANCH`   | previous commit target branch (sets only for `pull_request` and `pull_request_closed` events)                                                                                     | `main`                                                                                     |
```
Comment (pat-s) [inline docs/docs/20-usage/50-environment.md:105]:
```suggestion
| `CI_PREV_COMMIT_TARGET_BRANCH`   | previous commit target branch (set only for `pull_request` and `pull_request_closed` events)                                                                                     | `main`                                                                                     |
```

**Review by pat-s — COMMENTED** [review]
I can see arguments for both sides. I am not against a merge if this change gets properly documented, so that users can adjust their potential checks against empty/non-empty vars to now check for the existence.

While writing this, I realized that removing vars entirely makes checking a bit more cumbersome as one first needs to check for existence and then chain a check for a possible matching value.

@zc-devs We're currently curating issues for 3.0. If you want to get this into 3.0, please rebase and address the review in the next days.

**Comment by zc-devs** [comment]
I don't care, TBH.
Do maintainers want? Vote on this comment.
It doesn't make sense to merge, correct and write the docs, if someone disagree.

**Comment by pat-s** [comment]
You opened this PR, so I guess you care at least a little.

Nobody objects and I am also fine with it, so it is (still) up to you to rebase and get it in :)

**Comment by pat-s** [comment]
@zc-devs test failures

**Review by pat-s — APPROVED** [review]

**Pull request merged**